include/boost/math/quadrature/gauss_kronrod.hpp

```diff
@@ -82,138 +82,73 @@ class gauss_kronrod_detail;
 
 #endif
 
+#ifndef BOOST_HAS_FLOAT128
 template <class T>
 class gauss_kronrod_detail<T, 15, 0>
```

Comment:
Do you need to have the final template parameter be an integer or can you just use the `gauss_constant_category<T>::storage_type` as the final template parameter?

Reply:
Not quite, we now have 3 cases, two of which both use T as storage_type.